zulip/integrations/confluence/confluence.py

```diff
@@ -0,0 +1,110 @@
+rest api due to which there was a change in length in json data so i compare the
+length with database length if length is less  than update length then i update the
+database length and fetch the last update data from json response and send it to zulip
+stream.
+"""
+import os
+import sys
+import argparse
+import zulip
+import requests
+import json
+INSTRUCTIONS = r"""
+confluence-bot --config-file=~/.zuliprc --search="@nprnews,quantum physics"
+Send Confluence space and page  to a Zulip stream.
+To use this script:
+0. Use `Confluence rest api ` to  `Confluence `
+1. Set up Confluence authentication, as described below
+2. Set up a Zulip bot user and download its `.zuliprc` config file to e.g. `~/.zuliprc`
+3. Subscribe the bot to the stream that will receive Confluence updates (default stream: confluence)
+4. Test the script by running it manually, like this:
+5. Configure a crontab entry for this script. A sample crontab entry
+that will process spaces and pages every 5 minutes is:
+*/5 * * * * /usr/local/share/zulip/integrations/confluence/confluence [options]
+== Setting up Confluence authentications ==
+Run this on a personal or trusted machine, because your API key is
+visible to local users through the command line or config file.
+This bot uses OAuth to authenticate with Confluence. Please create a
+~/.zulip_confluencerc with the following contents:
+[confluence]
+username = example@.com
+api_token = xxxxxxxxxxxx
+In order to obtain a consumer key & secret, you must register a
+new application under your Confluence account:
+1 Log in to https://id.atlassian.com.
+2 Click API tokens, then Create API token.
+3 Use Copy to clipboard, and paste the token to your script, or elsewhere:
+"""
+f = open("paras.txt", "w+")
+f.write('0')
+f.close()
+def confluence_api() -> None:
+    # Use rest api to fetch data from confluence here
+    # In authentication we provide confluence api token and then username
+    params = ('type', 'page')
+    response = requests.get('https://paras112.atlassian.net/wiki/rest/api/content',
+                            params=params, auth=('example@gmail.com', 'XXXXXXXXXXXXX'))
+    res = response.json()
+    # length of json data
+    length_api = len(res['results'])
+    # Fetch value from database models
+    f = open("paras.txt", "r")
+    length_confluence_data = int(f.read())
+    # compare the lenth of json from database to the new length of json which is fetch from api
+    # if length of fetch json from api  is greater than save length then we call zulip api
+    if (length_api > length_confluence_data):
+        test1_json = (res['results'][length_api-1]['_links']['webui'])
+        title = res['results'][length_api-1]['title']
+        get_spaces_and_pages = spacepage(test1_json)
+        format_page_and_string = "New {} is created with title {}".format(get_spaces_and_pages, title)
+        client = zulip.Client(config_file = "/home/paras/Downloads/zuliprc")
```

Comment:
I'm not sure if you should be adding paths relative to your local environment here

Reply:
Agree with @mayankmadan

Reply:
I don't know  how to use path relatives here in my code ? Please help me ?

Reply:
I think a similar approach to twitter bot can be used here. i.e Passing the path to the config file as an argument to the script so that its not hardcoded

Reply:
Yep, I would recommend imitating the Twitter integration as much as possible. :)

Reply:
Okay I agree Twitter bot is great and we should follow that approach  Thanks .